### DISCUSSION TITLE
Feature Request: Framework-Agnostic wagmi Library 🚀

### DISCUSSION THREAD

**Discussion by denniarems** [wevm/wagmi#3275] [Ideas]
### Feature Request: Framework-Agnostic wagmi Library 🚀

**Yo, wagmi devs and community!**

First off, big ups for all the hard work y'all have been putting into this project. It's legit helping us vibe with the blockchain in ways we never thought possible. But hey, I've got a fresh idea that could make wagmi even more bussin'.

#### Here's the tea: 🍵

**Idea: Let's make wagmi Framework-Agnostic!**

Right now, wagmi is tight with React, and no cap, that's cool and all. But imagine how lit it would be if we could make wagmi jive with other frameworks too. There's a whole world beyond React that's just waiting to yeet into the blockchain space!

#### Why This Slaps: 🔥

- **More Inclusive:** We got peeps working with Vue, Angular, Svelte, and even those dabbling with newer tech. Making wagmi more inclusive to other frameworks? That's a major glow-up.
- **Flexibility is Key:** It's all about giving devs the freedom to choose their tools. More choices, more creativity.
- **Boost the Community:** By expanding the ecosystem, we're not just building a tool; we're building a community. And that's what wagmi's all about, right?

#### What It Might Look Like:

- **Modular Design:** Refactor the core to be UI framework-agnostic. Keep it clean, keep it fresh.
- **Documentation & Examples:** We gotta make sure we're not leaving anyone in the dark. Clear docs and examples for different frameworks? Yes, please.
- **Community Collaboration:** Let's get the input from devs who stan other frameworks. They know their stuff and can help make this feature a total banger.

**Comment by tmm** [comment]
Core is already framework agnostic. As part of [Wagmi v2](https://github.com/wevm/wagmi/discussions/3068), query and mutation functions and keys are pulled into the core as well. You can see these changes on the [`beta`](https://github.com/wevm/wagmi/tree/beta) branch. The `wagmi` (React) package is now mostly core functions and types.

We'd love to support other frameworks at some point in the future after Wagmi v2 is released. In the meantime, you are welcome to use `@wagmi/core` with any framework (lots of folks already do this) and there are a few community efforts out there on GitHub you can check out.
